massa-versioning/src/versioning.rs

```diff
@@ -179,30 +179,55 @@ pub struct Advance {
     pub now: MassaTime,
 }
 
-// Need Ord / PartialOrd so it is properly sorted in BTreeMap
+impl PartialEq for Advance {
+    fn eq(&self, other: &Self) -> bool {
+        self.start_timestamp == other.start_timestamp
+            && self.timeout == other.timeout
+            && self.threshold == other.threshold
+            && self.now == other.now
+            && self.activation_delay == other.activation_delay
+    }
+}
+
+impl Eq for Advance {}
+
+// A Lightweight version of 'Advance' (used in MipState history)
+#[derive(Clone, Debug)]
+pub struct AdvanceLW {
```

Comment:
It's a bit heavy to create a new struct just for this cleanup IMO. If you can find a way to just use the now it would be great ;)

Reply:
Try to use only now but it breaks a lot of things. I think this is the best solution AFAIK